redbot/core/commands/requires.py

```diff
@@ -31,7 +32,7 @@
     from .commands import Command
     from .context import Context
 
-    _CommandOrCoro = TypeVar("_CommandOrCoro", Callable[..., Awaitable[Any]], Command)
+    _CommandOrCoro = TypeVar("_CommandOrCoro", bound=Command)  # lie to mypy
```

Comment:
This type is supposed to represent both `commands.Command` and `Callable[..., Awaitable[Any]]]`, otherwise developers are gonna run into issues when they use check decorators before `@commands.command()` so I think this shouldn't be left like that unless I'm missing something here.

Reply:
I've put this back for now, but I'm not entirely sure we should be encouraging decorator orders which don't stack logically as these end up doing assignments to function objects which gets much more difficult to write good typing for.